File: core/src/main/java/y2k/joyreactor/platform/Platform.java
package y2k.joyreactor.platform;

import java.io.File;

/**
 * Created by y2k on 29/09/15.
 */
public abstract class Platform {

    public static Platform Instance;

    public abstract File getCurrentDirectory();

    public abstract Navigation getNavigator();

    public abstract byte[] loadFromBundle(String name, String ext);
}

File: core/src/main/kotlin/y2k/joyreactor/platform/Platform.kt
package y2k.joyreactor.platform

import com.j256.ormlite.support.ConnectionSource
import rx.Observable

import java.io.File

/**
 * Created by y2k on 29/09/15.
 */
abstract class Platform {

    abstract val currentDirectory: File

    abstract val navigator: NavigationService

    abstract fun loadFromBundle(name: String, ext: String): ByteArray

    abstract fun saveToGallery(imageFile: File): Observable<*>

    abstract fun buildConnection(file: File): ConnectionSource

    companion object {

        lateinit var instance: Platform
    }
}